### PR TITLE
[AutomaticAssessmentConfiguration] Update Binding to Xcode 12 beta 5

### PR DESCRIPTION
**Pull request by tj-devel709**

### CODE REVIEW

File: src/automaticassessmentconfiguration.cs [hunk 1]
```diff
@@ -21,13 +22,76 @@ namespace AutomaticAssessmentConfiguration {
...
+		Punctuation = 1 << 1,
+	}
 	
 	[Unavailable (PlatformName.MacCatalyst)]
 	[Mac (10,15,4), iOS (13,4)]
 	[Advice ("This API is not available when using UIKit on macOS.")]
 	[BaseType (typeof (NSObject))]
 	interface AEAssessmentConfiguration : NSCopying {
 
+		[Unavailable (PlatformName.MacCatalyst)]
```
Comment (spouliot) [inline src/automaticassessmentconfiguration.cs:42]: not needed, it's already defined on the type itself

File: src/automaticassessmentconfiguration.cs [hunk 2]
```diff
@@ -21,13 +22,76 @@ namespace AutomaticAssessmentConfiguration {
...
+		[NoMac, iOS (14, 0)]
+		[Advice ("This API is not available when using UIKit on macOS.")]
+		[Export ("autocorrectMode")]
+		AEAutocorrectMode AutocorrectMode { get; set; }
+
+		[Unavailable (PlatformName.MacCatalyst)]
```
Comment (spouliot) [inline src/automaticassessmentconfiguration.cs:48]: same

File: src/automaticassessmentconfiguration.cs [hunk 3]
```diff
@@ -21,13 +22,76 @@ namespace AutomaticAssessmentConfiguration {
...
+		[NoMac, iOS (14, 0)]
+		[Advice ("This API is not available when using UIKit on macOS.")]
+		[Export ("allowsSpellCheck")]
+		bool AllowsSpellCheck { get; set; }
+
+		[Unavailable (PlatformName.MacCatalyst)]
```
Comment (spouliot) [inline src/automaticassessmentconfiguration.cs:54]: same

File: src/automaticassessmentconfiguration.cs [hunk 4]
```diff
@@ -21,13 +22,76 @@ namespace AutomaticAssessmentConfiguration {
...
+		[NoMac, iOS (14, 0)]
+		[Advice ("This API is not available when using UIKit on macOS.")]
+		[Export ("allowsPredictiveKeyboard")]
+		bool AllowsPredictiveKeyboard { get; set; }
+
+		[Unavailable (PlatformName.MacCatalyst)]
```
Comment (spouliot) [inline src/automaticassessmentconfiguration.cs:60]: same

File: src/automaticassessmentconfiguration.cs [hunk 5]
```diff
@@ -21,13 +22,76 @@ namespace AutomaticAssessmentConfiguration {
...
+		[NoMac, iOS (14, 0)]
+		[Advice ("This API is not available when using UIKit on macOS.")]
+		[Export ("allowsKeyboardShortcuts")]
+		bool AllowsKeyboardShortcuts { get; set; }
+
+		[Unavailable (PlatformName.MacCatalyst)]
```
Comment (spouliot) [inline src/automaticassessmentconfiguration.cs:66]: same

File: src/automaticassessmentconfiguration.cs [hunk 6]
```diff
@@ -21,13 +22,76 @@ namespace AutomaticAssessmentConfiguration {
...
+		[NoMac, iOS (14, 0)]
+		[Advice ("This API is not available when using UIKit on macOS.")]
+		[Export ("allowsActivityContinuation")]
+		bool AllowsActivityContinuation { get; set; }
+
+		[Unavailable (PlatformName.MacCatalyst)]
```
Comment (spouliot) [inline src/automaticassessmentconfiguration.cs:72]: same

File: src/automaticassessmentconfiguration.cs [hunk 7]
```diff
@@ -21,13 +22,76 @@ namespace AutomaticAssessmentConfiguration {
...
+		[NoMac, iOS (14, 0)]
+		[Advice ("This API is not available when using UIKit on macOS.")]
+		[Export ("allowsDictation")]
+		bool AllowsDictation { get; set; }
+
+		[Unavailable (PlatformName.MacCatalyst)]
```
Comment (spouliot) [inline src/automaticassessmentconfiguration.cs:78]: same

File: src/automaticassessmentconfiguration.cs [hunk 8]
```diff
@@ -21,13 +22,76 @@ namespace AutomaticAssessmentConfiguration {
...
+		[NoMac, iOS (14, 0)]
+		[Advice ("This API is not available when using UIKit on macOS.")]
+		[Export ("allowsAccessibilitySpeech")]
+		bool AllowsAccessibilitySpeech { get; set; }
+
+		[Unavailable (PlatformName.MacCatalyst)]
```
Comment (spouliot) [inline src/automaticassessmentconfiguration.cs:84]: same

File: src/automaticassessmentconfiguration.cs [hunk 9]
```diff
@@ -21,13 +22,76 @@ namespace AutomaticAssessmentConfiguration {
...
+		[NoMac, iOS (14, 0)]
+		[Advice ("This API is not available when using UIKit on macOS.")]
+		[Export ("allowsPasswordAutoFill")]
+		bool AllowsPasswordAutoFill { get; set; }
+
+		[Unavailable (PlatformName.MacCatalyst)]
```
Comment (spouliot) [inline src/automaticassessmentconfiguration.cs:90]: same

File: src/automaticassessmentconfiguration.cs [hunk 10]
```diff
@@ -3,6 +3,7 @@
 //
 // Authors:
 //	Alex Soto <alexsoto@microsoft.com>
+//	TJ Lambert <TJ.Lambert@microsoft.com>
 //
 // Copyright (c) Microsoft Corporation.
```
Comment (whitneyschmidt) [inline src/automaticassessmentconfiguration.cs:8]:
```suggestion
// Copyright 2020, Microsoft Corporation.
```

**Comment by tj-devel709** [comment]
I fixed the suggestions and when I tried to push the changes, I got a git error about the tip of my current branch being behind it's remote counterpart and it suggesting a git pull. I had already fetched xamarin and rebased before but did the suggested git pull and when I pushed, tests/xharness/Jenkins/Jenkins.cs had some changes. Is this because someone else has made those changes in the last few hours?

**Comment by whitneyschmidt** [comment]
> I fixed the suggestions and when I tried to push the changes, I got a git error about the tip of my current branch being behind it's remote counterpart and it suggesting a git pull. I had already fetched xamarin and rebased before but did the suggested git pull and when I pushed, tests/xharness/Jenkins/Jenkins.cs had some changes. Is this because someone else has made those changes in the last few hours?

You can check the commits on xcode12 to see whether anything new was committed :) It looks like UIKit changes were merged 7 min ago.

File: src/automaticassessmentconfiguration.cs [hunk 11]
```diff
@@ -3,8 +3,9 @@
 //
 // Authors:
 //	Alex Soto <alexsoto@microsoft.com>
+//	TJ Lambert <TJ.Lambert@microsoft.com>
 //
-// Copyright (c) Microsoft Corporation.
+// Copyright (2020) Microsoft Corporation.
```
Comment (dalexsoto) [inline src/automaticassessmentconfiguration.cs:8]: Do not change copyright headers, this is correct as it was

```suggestion
// Copyright (c) Microsoft Corporation.
```

File: src/automaticassessmentconfiguration.cs [hunk 12]
```diff
@@ -21,13 +22,67 @@ namespace AutomaticAssessmentConfiguration {
...
+	}
 	
 	[Unavailable (PlatformName.MacCatalyst)]
 	[Mac (10,15,4), iOS (13,4)]
 	[Advice ("This API is not available when using UIKit on macOS.")]
 	[BaseType (typeof (NSObject))]
 	interface AEAssessmentConfiguration : NSCopying {
 
+		[NoMac, iOS (14, 0)]
+		[Advice ("This API is not available when using UIKit on macOS.")]
```
Comment (dalexsoto) [inline src/automaticassessmentconfiguration.cs:43]: This is not needed as the parent class already holds this advice

```suggestion
		[Advice ("This API is not available when using UIKit on macOS.")]
```

Comment (dalexsoto) [reply src/automaticassessmentconfiguration.cs:43]: same for all the ones below within `AEAssessmentConfiguration`

**Comment by tj-devel709** [comment]
My branch ended up getting polluted so with Alex's help, I am closing this PR and creating a new PR with a clean branch! :) Sorry for any inconvenience!

**Pull request closed without merging**